internal/client/api/backend.go

```diff
@@ -352,3 +353,51 @@ type Backend[
 	// TODO: implement UsageInfo if we require it
 	// UsageInfo() *UsageInfo
 }
+
+// StorageProvider provides access to storage primitives
+type StorageProvider[H runtime.Hash, Hasher runtime.Hasher[H]] interface {
+	// Storage returns the value under the key in that block, given a blocks hash and a key.
+	Storage(hash H, key statemachine.StorageKey) (statemachine.StorageValue, error)
```

Comment:
please use the correct primitives during this development phase.  We can consolidate the primitives after if it really doesn't make sense to differentiate between `storage.StorageData` and `statemachine.StorageValue`.